install_config/configuring_extended_ldap_attributes.adoc

```diff
@@ -0,0 +1,175 @@
+= Configuring Extended LDAP Attributes
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+== Overview
+This topic builds upon xref:../install_config/ssd_for_ldap_failover.adoc[Setting
+up SSSD for LDAP Failover] and
+xref:../install_config/configuring_form_based_authentication.adoc[Configuring
+Form-Based Authentication] and focused on configuring extended Lightweight
+Directory Access Protocol (LDAP) attributes.
+
+[[configuring-extended-ldap-attributes-prerequisites]]
+== Prerequisites
+
+* SSSD 1.12.0 or later. This is available on Red Hat Enterprise Linux 7.0 and
+later.
+* mod_lookup_identity 0.9.4 or later.
+** This is not yet available on any version of Red Hat Enterprise Linux 7, but the
+Red Hat Package Managers (RPMs) for this platform are
+https://copr.fedorainfracloud.org/coprs/adelton/identity_demo/[available for
+upstream] until they arrive in Red Hat Enterprise Linux.
+
+[[configuring-extended-ldap-attributes-config-sssd]]
+== Configuring SSSD
+You need to ask System Security Services Daemon (SSSD) to look up attributes in LDAP that it normally doesn’t care
+about for simple system-login use-cases. In the OpenShift case, there’s really
+only one such attribute: email. So we need to modify the [domain/DOMAINNAME]
+section of /etc/sssd/sssd.conf on the authenticating proxy and add this
+attribute:
+
+----
+[domain/example.com]
+...
+ldap_user_extra_attrs = mail
+----
+
+ Next, tell SSSD that it is acceptable for this attribute to be retrieved by
+Apache. Add the following two lines to the *[ifp]* section of
+*_/etc/sssd/sssd.conf_*:
+
+====
+----
+[ifp]
+user_attributes = +mail
+allowed_uids = apache, root
+----
+====
+
+Restart SSSD and test this configuration:
+
+====
+----
+# systemctl restart sssd.service
+
+# getent passwd <username>
+username:*:12345:12345:Example User:/home/username:/usr/bin/bash
+
+# gdbus call \
+        --system \
+        --dest org.freedesktop.sssd.infopipe \
+        --object-path /org/freedesktop/sssd/infopipe/Users/example_2ecom/12345 \
+        --method org.freedesktop.DBus.Properties.Get \
+        "org.freedesktop.sssd.infopipe.Users.User" "extraAttributes"
+(<{'mail': ['username@example.com']}>,)
+----
+====
+
+[[configuring-extended-ldap-attributes-config-apache]]
+== Configuring Apache
+
+Now that SSSD is set up and successfully serving extended attributes, configure
+the web server to ask for them and to insert them in the correct places.
+
+First, install and enable the *mod_lookup_identity* module for Apache:
+
+====
+----
+# yum -y install mod_lookup_identity
+----
+====
+
+Second, enable the module so that Apache will load it. Modify
```

Comment:
@sgallagher - yep, I'd agree with that. Thanks for the help there. We just try to stay in active voice as much as possible for tech writing, instead of passive voice or vague future tense.